Lib/test/test_xml_etree.py

```diff
@@ -2423,6 +2423,14 @@ def test_39495_treebuilder_start(self):
         self.assertRaises(TypeError, ET.TreeBuilder().start, "tag")
         self.assertRaises(TypeError, ET.TreeBuilder().start, "tag", None)
 
+    def test_issue123213_correct_extend_exception(self):
+        # Does not hide the internal exception when extending the element
+        self.assertRaises(ZeroDivisionError, ET.Element('tag').extend,
+                          (1/0 for i in range(2)))
+
+        # Still raises the TypeError when extending with a non-iterable
+        self.assertRaises(TypeError, ET.Element('tag').extend, None)
```

Comment:
Just one more test case maybe: use a generator of the form `(f() for i in range(2))` where `f` raises `TypeError("hello")`. Then check that this TypeError message is the correct one.

Reply:
Good idea